server/app.es6

```diff
@@ -1,73 +1,83 @@
-
-// server creation ----------------------------------------------------------
+let tempOutfitter = require('./temp-outfitter.es6');
+let util = require('./util.es6');
+let vcapServices = require('./vcap-services.es6');
 
 let app = express();
-app.use(bodyParser.json());
 
-// middleware that will add the Access-Control-Allow-Origin header to everything
-app.use(function(req, res, next) {
-  var origin = req.headers.origin;
-  if (origin === 'http://localhost:4200' || origin === 'http://localhost:49152') {
-    res.set('Access-Control-Allow-Origin', origin);
-  } else {
-    res.set('Access-Control-Allow-Origin', 'https://fs-intake-staging.app.cloud.gov');
-  }
-  res.set('Access-Control-Allow-Credentials', true);
-  next();
-});
+/* Use helmet for increased security. */
+app.use(helmet());
+
+/* Parse request bodies as JSON. */
+app.use(bodyParser.json());
 
-// set these headers on all of the OPTIONS preflight responses
 app.options('*', function(req, res) {
+  res.set('Access-Control-Allow-Origin', vcapServices.intakeClientBaseUrl);
+  res.set('Access-Control-Allow-Credentials', true);
   res.set('Access-Control-Allow-Headers', 'accept, content-type');
   res.set('Access-Control-Allow-Methods', 'GET, PUT, POST, DELETE, OPTIONS, PATCH');
   res.send();
 });
 
-app.use('/docs/api', express.static('docs/api'));
-
-// Endpoints
-
-// POST /permits/applications/special-uses/noncommercial/
-// creates a new noncommercial application
-app.post('/permits/applications/special-uses/noncommercial', noncommercial.createNoncommercialTempApp);
-
-// POST /permits/applications/special-uses/temp-outfitters
-// creates a new temp outfitter application
-app.post('/permits/applications/special-uses/temp-outfitters', tempOutfitter.createTempOutfitterApp);
-
-// POST /permits/applications/special-uses/temp-outfitters/file
-// Handles temp outfitter file upload and invokes streamToS3 function
-app.post('/permits/applications/special-uses/temp-outfitters/file', tempOutfitter.streamToS3.array('file', 1), tempOutfitter.createAppFile);
-
-// PUT /permits/applications/special-uses/noncommercial/:tempControlNumber
-// updates an existing noncommercial application
-// may not be able to update everything wholesale due to possible field audit requirements
-app.put('/permits/applications/special-uses/noncommercial/:id', noncommercial.updateApp);
-
-// GET /permits/applications/special-uses/noncommercial/:tempControlNumber
-app.get('/permits/applications/special-uses/noncommercial/:id', noncommercial.getApp);
-
-// GET /permits/applications/special-uses/temp-outfitters/:tempControlNumber
-app.get('/permits/applications/special-uses/temp-outfitters/:id', tempOutfitter.getApp);
+/* Authentication middleware. */
+let authenticator = function(req, res, next) {
+  res.set('Access-Control-Allow-Origin', vcapServices.intakeClientBaseUrl);
+  res.set('Access-Control-Allow-Credentials', true);
+  let user = auth(req);
+  if (process.env.PLATFORM === 'CircleCI') {
```

Comment:
I wonder if its worth generalizing this to just `CI` or `continuous-integration` so that its not so dependent to what we are currently using.